pkg/logs/tailers/socket/tailer.go

```diff
@@ -67,7 +69,19 @@ func (t *Tailer) forwardMessages() {
 	}()
 	for output := range t.decoder.OutputChan {
 		if len(output.GetContent()) > 0 {
-			t.outputChan <- message.NewMessageWithSource(output.GetContent(), message.StatusInfo, t.source, output.IngestionTimestamp)
+			origin := message.NewOrigin(t.source)
+			remoteAddress := t.Conn.RemoteAddr()
+			copiedTags := make([]string, len(t.source.Config.Tags))
+			copy(copiedTags, t.source.Config.Tags)
+			if remoteAddress != nil {
+				addressWithPort := t.Conn.RemoteAddr().String()
+				lastColonIndex := strings.LastIndex(addressWithPort, ":")
+				ipAddress := addressWithPort[:lastColonIndex]
```

Comment:
index will be `-1` if `:` not found. which will cause index out of bounds error (which is why some tests are failing right now)